adr/ADR-33.md

```diff
@@ -0,0 +1,257 @@
+in the way as we do for `$JS`, in order to enable targetting tools to work
+across accounts.
+
+The initial _verbs_ supported by the service include:
+
+- `PING`
+- `STATS`
+- `INFO`
+- `SCHEMA`
+
+Using the above verbs, it becomes possible to build a service subject hierarchy
+like:
+
+`$SRV.PING|STATS|INFO|SCHEMA` - pings and retrieves status for all services
+`$SRV.PING|STATS|INFO|SCHEMA.<name>` - pings or retrieves status for all
+services having the specified name `$SRV.PING|STATS|INFO|SCHEMA.<name>.<id>` -
+pings or retrieves status of a particular service instance
+
+Services should respond to:
+
+- All service requests
+- All service requests that match their `name`
+- All services requests that match their `name` and `id`
+
+### Standard Field
+
+All discovery and status responses contain the following fields:
+
+```typescript
+    /**
+    * The kind of the service reporting the status
+    */
+    name: string,
+    /**
+    * The unique ID of the service reporting the status
+    */
+    id: string,
+    /**
+    * The version of the service
+    */
+    version: string
+}
+```
+
+### INFO
+
+Returns a JSON having the following structure:
+
+```typescript
+{
+    name: string,
+    id: string,
+    version: string,
+    /**
+    * Description for the service
+    */
+    description: string,
+    /**
+     * Version of the service
+     */
+    version: string,
+    /**
+     * Subject where the service can be invoked
+     */
+    subject: string
+}
+```
+
+All the fields above map 1-1 to the metadata provided when the service was
+created. Note that `subject` is the subject that the service is listening for
+requests on.
+
+### PING
+
+Returns the following schema (the standard response fields)
+
+```typescript
+{
+    name: string,
+    id: string,
+    version: string,
+}
+```
+
+The intention of `PING` is for clients to calculate RTT to a service and
+discover services.
+
+### SCHEMA
+
+Returns a JSON having the following structure. Note that the `schema` struct is
+only returned if the `schema` was specified when created.
+
+```typescript
+{
+    name: string,
+    id: string,
+    version: string,
+    /**
+     * The schema specified when the service was created
+     */
+    schema?: {
+        /**
+         * A string or URL
+         */
+        request: string,
+        /**
+         * A string or URL
+         */
+        response: string
+    };
+}
+```
+
+### STATS
+
+```typescript
+{
+    name: string,
+    id: string,
+    version: string,
+    /**
+    * The number of requests received by the endpoint
+    */
+    num_requests: number;
+    /**
+    * Number of errors that the endpoint has raised
+    */
+    num_errors: number;
+    /**
+    * If set, the last error triggered by the endpoint
+    */
+    last_error?: Error;
+    /**
+    * A field that can be customized with any data as returned by stats handler see {@link ServiceConfig}
+    */
+    data?: unknown;
```

Comment:
This should just be a string. Would make it much easier to handle all data because, like a message, it's just a payload